File: Linq2TwitterDemos_Console/ListDemos.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using LinqToTwitter;

namespace Linq2TwitterDemos_Console
{
    class ListDemos
    {
        internal static async Task RunAsync(TwitterContext twitterCtx)
        {
            char key;

            do
            {
                ShowMenu();

                key = Console.ReadKey(true).KeyChar;

                switch (key)
                {
                    case '0':
                        Console.WriteLine("\n\tGetting Lists...\n");
                        await GetListsForUser(twitterCtx);
                        break;
                    case 'q':
                    case 'Q':
                        Console.WriteLine("\nReturning...\n");
                        break;
                    default:
                        Console.WriteLine(key + " is unknown");
                        break;
                }

            } while (char.ToUpper(key) != 'Q');
        }

        static async Task GetListsForUser(TwitterContext twitterCtx)
        {
            var lists =
                await
                    (from list in twitterCtx.List
                     where list.Type == ListType.Lists &&
                           list.ScreenName == "Linq2Tweeter"
                     select list)
                    .ToListAsync();

            lists.ForEach(list => Console.WriteLine("Slug: " + list.SlugResult));
        }

        static void ShowMenu()
        {
            Console.WriteLine("\nList Demos - Please select:\n");

            Console.WriteLine("\t 0. Get Lists for User");
            Console.WriteLine();
            Console.WriteLine("\t Q. Return to Main menu");
        }
    }
}

File: Samples/LinqToTwitter6/Console/ConsoleDemos.VisualBasic/ConsoleDemos.VisualBasic/ListDemos.vb
Imports LinqToTwitter

Friend Class ListDemos
    Friend Shared Async Function RunAsync(twitterCtx As TwitterContext) As Task
        Dim screenName = "Linq2Twitr"

        Dim lists =
            Await _
            (From list In twitterCtx.List
             Where list.Type = ListType.List AndAlso
                   list.ScreenName = screenName
             Select list) _
            .ToListAsync()

        If lists IsNot Nothing Then
            lists.ForEach(Sub(list) Console.WriteLine("Slug: " + list.SlugResponse))
        End If
    End Function
End Class
